### PR TITLE
refactor!: Use BasePropagator interface in vertexing

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR untemplates the ImpactParameterEstimator, HelicalTrackLinearizer and NumericalTrackLinearizer

Part of:
- #2842 

Blocked by:
- #2881

### CODE REVIEW

**Comment by paulgessinger** [comment]
This is ready now, @andiwand

File: Core/include/Acts/Vertexing/HelicalTrackLinearizer.hpp [hunk 1]
```diff
@@ -38,46 +41,18 @@ namespace Acts {
...
-  struct State {
-    /// @brief The state constructor
-    ///
-    /// @param fieldCacheIn The magnetic field cache
-    State(MagneticFieldProvider::Cache fieldCacheIn)
-        : fieldCache(std::move(fieldCacheIn)) {}
-    /// Magnetic field cache
-    MagneticFieldProvider::Cache fieldCache;
-  };
+  struct State {};
```
Comment (andiwand) [inline Core/include/Acts/Vertexing/HelicalTrackLinearizer.hpp:47]: is this used somewhere or can we drop it completely?

File: Core/include/Acts/Vertexing/HelicalTrackLinearizer.ipp [hunk 2]
```diff
@@ -39,14 +39,19 @@ Acts::Result<Acts::LinearizedTrack> Acts::
       Direction::fromScalarZeroAsPositive(intersection.pathLength());
 
   // Propagate to the PCA of the reference point
-  auto result = m_cfg.propagator->propagate(params, perigeeSurface, pOptions);
-  if (!result.ok()) {
-    return result.error();
+  // auto result = m_cfg.propagator->propagate(params, perigeeSurface,
+  // pOptions); if (!result.ok()) { return result.error();
+  // }
+  // const auto& endParams = *result->endParameters;
```
Comment (andiwand) [inline Core/include/Acts/Vertexing/HelicalTrackLinearizer.ipp:45]: dead code?

File: Core/include/Acts/Vertexing/NumericalTrackLinearizer.ipp [hunk 3]
```diff
@@ -124,12 +123,12 @@ Acts::NumericalTrackLinearizer<propagator_t, propagator_options_t>::
...
     // Propagate to the new PCA and extract Perigee parameters
-    auto newResult = m_cfg.propagator->propagate(wiggledCurvilinearParams,
-                                                 perigeeSurface, pOptions);
+    auto newResult = m_cfg.propagator->propagateToSurface(
+        wiggledCurvilinearParams, perigeeSurface, pOptions);
     if (!newResult.ok()) {
       return newResult.error();
     }
-    newPerigeeParams = (*newResult->endParameters).parameters();
+    newPerigeeParams = (*newResult).parameters();
```
Comment (andiwand) [inline Core/include/Acts/Vertexing/NumericalTrackLinearizer.ipp:131]:
```suggestion
    newPerigeeParams = newResult->parameters();
```

**Review by andiwand — DISMISSED** [review]
yes please!

File: Core/include/Acts/Vertexing/HelicalTrackLinearizer.hpp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/Vertexing/HelicalTrackLinearizer.hpp:47]: Hm, it's still part of the `LinearizerConcept`, which is about to become obsolete anyway. I can remove the State from the concept here. The `NumericalTrackLinearizer` does not fulfill that concept anyway.

**Comment by paulgessinger** [comment]
Updated @andiwand

**Review by andiwand — DISMISSED** [review]

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [9eac779e07d8a4a34e94eb4afdb48b3c78a7393f]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/36155431) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6900576)!

**Pull request merged**